### PR TITLE
Faster/Better ens. B processing/recentering

### PR DESCRIPTION
**Pull request by guillaumevernieres**
## Description
Features added:
- replaced multiple steps in the workflow by 1 executable, limiting the io to a bare minimum and decreasing the job wall-time from close to 30mn donw to ~10mn. Still too slow, but better!
- addition of background error related diagnostics (explained variance by the physical balance, unbalanced bkg-err, ...)
- the new field are plotted in the verify task

waiting for [oops/pull/2340](https://github.com/JCSDA-internal/oops/pull/2340)

**DON'T** trigger the CI for this draft PR just yet, it's pointing to `jcsda-internal` for `oops`.

### CODE REVIEW

File: scripts/exgdas_global_marine_analysis_post.py [hunk 1]
```diff
@@ -59,28 +59,33 @@ def list_all_files(dir_in, dir_out, wc='*', fh_list=[]):
...
-                       os.path.join(com_ocean_analysis, f'{RUN}.t{cyc}z.ocn.bkgerr_stddev.nc')])
-post_file_list.append([os.path.join(anl_dir, 'static_ens', f'ice.orig_ens_stddev.incr.{bdate}.nc'),
-                       os.path.join(com_ocean_analysis, f'{RUN}.t{cyc}z.ice.bkgerr_stddev.nc')])
-
-# Copy the ice and ocean increments
-post_file_list.append([os.path.join(anl_dir, 'Data', f'ocn.3dvarfgat_pseudo.incr.{mdate}.nc'),
-                       os.path.join(com_ocean_analysis, f'{RUN}.t{cyc}z.ocn.incr.nc')])
-post_file_list.append([os.path.join(anl_dir, 'Data', f'ice.3dvarfgat_pseudo.incr.{mdate}.nc'),
-                       os.path.join(com_ocean_analysis, f'{RUN}.t{cyc}z.ice.incr.nc')])
+domains = ['ocn', 'ice']
```
Comment (CoryMartin-NOAA) [inline scripts/exgdas_global_marine_analysis_post.py:62]: suggest not using `domains` as that makes me think `global` or `GulfOfMexico`, etc.

How about `components`?

File: scripts/exgdas_global_marine_analysis_post.py [hunk 2]
```diff
@@ -59,28 +59,33 @@ def list_all_files(dir_in, dir_out, wc='*', fh_list=[]):
...
+for domain in domains:
+    # Copy of the diagonal of the background error for the cycle
+    post_file_list.append([os.path.join(anl_dir, 'static_ens', f'{domain}.bkgerr_stddev.incr.{bdate}.nc'),
```
Comment (CoryMartin-NOAA) [inline scripts/exgdas_global_marine_analysis_post.py:65]: We should talk tomorrow about using Jinja YAML templates and not having this be scripted. Not important for this PR, but future.

File: utils/soca/gdas_ens_handler.h [hunk 3]
```diff
@@ -0,0 +1,263 @@
+#pragma once
+
+#include <netcdf>
```
Comment (CoryMartin-NOAA) [inline utils/soca/gdas_ens_handler.h]: what is netCDF used for?

**Review by CoryMartin-NOAA — COMMENTED** [review]
Just some minor comments/questions

**Comment by guillaumevernieres** [comment]
@ShastriPaturi and @AndrewEichmann-NOAA , when you have time, please review.

File: scripts/exgdas_global_marine_analysis_post.py (hunk 1 above)
Comment (guillaumevernieres) [reply scripts/exgdas_global_marine_analysis_post.py:62]: multi ... domain ... ufo ... 
Nope, it's staying!

File: scripts/exgdas_global_marine_analysis_post.py (hunk 2 above)
Comment (guillaumevernieres) [reply scripts/exgdas_global_marine_analysis_post.py:65]: ok

File: utils/soca/gdas_ens_handler.h (hunk 3 above)
Comment (guillaumevernieres) [reply utils/soca/gdas_ens_handler.h]: good catch ... not used.

File: scripts/exgdas_global_marine_analysis_post.py (hunk 1 above)
Comment (AndrewEichmann-NOAA) [reply scripts/exgdas_global_marine_analysis_post.py:62]: kingdom, phylum

File: scripts/exgdas_global_marine_analysis_vrfy.py [hunk 4]
```diff
@@ -41,6 +41,61 @@
 HOMEgfs = os.getenv('HOMEgfs')
 
 
+#######################################
+# recentering error
+#######################################
+string = 'recentering_error'
```
Comment (AndrewEichmann-NOAA) [inline scripts/exgdas_global_marine_analysis_vrfy.py]: "string" is a little generic

**Review by AndrewEichmann-NOAA — DISMISSED** [review]

**Review by ShastriPaturi — DISMISSED** [review]
Looks good

File: scripts/exgdas_global_marine_analysis_post.py (hunk 2 above)
Comment (guillaumevernieres) [reply scripts/exgdas_global_marine_analysis_post.py:65]: created issue here: https://github.com/NOAA-EMC/GDASApp/issues/612

File: utils/soca/gdas_ens_handler.h (hunk 3 above)
Comment (guillaumevernieres) [reply utils/soca/gdas_ens_handler.h]: removed netcdf in one of the latest commits

File: scripts/exgdas_global_marine_analysis_post.py (hunk 1 above)
Comment (guillaumevernieres) [reply scripts/exgdas_global_marine_analysis_post.py:62]: Ignoring @CoryMartin-NOAA 's comment!!! (bad bad G)

File: scripts/exgdas_global_marine_analysis_vrfy.py (hunk 4 above)
Comment (guillaumevernieres) [reply scripts/exgdas_global_marine_analysis_vrfy.py]: removed string, no clue what the other stuff means @AndrewEichmann-NOAA

**Comment by guillaumevernieres** [comment]
I can't remember the last time the ci was successful the first time around! I'll take it :)
Can I have a re-review @CoryMartin-NOAA , @ShastriPaturi and @AndrewEichmann-NOAA ?

**Review by AndrewEichmann-NOAA — APPROVED** [review]
looks reasonable

**Pull request merged**